### PR TITLE
Initial refactor

### PR DESCRIPTION
**Pull request by zeryx**
fully functional release, needs docs and and maybe :eyes:  at how the "blobbing" mechanic works wrt interfering anomalies with varying sigma threshold.
Functional though

### CODE REVIEW

**Comment by besirkurtulmus** [comment]
Other than the commented line, LGTM.

Really like the code. It's super clean and concise!

File: src/OpenAnomalyDetection_test.py [hunk 1]
```diff
@@ -4,16 +4,17 @@
 
 def test_detect():
     input = dict()
-    input['data_path'] = 'data://TimeSeries/GenerativeForecasting/formatted_data_rossman_10.json'
-    input['model_input_path'] = 'file://tmp/rossman_0.1.0.zip'
-    input['graph_save_path'] = 'file://tmp/graph_file.png'
-    input['max_sigma'] = 2.0
-    input['variable_index'] = 4
+    input['data_path'] = 'data://TimeSeries/GenerativeForecasting/m4_daily.json'
+    # input['model_input_path'] = 'file://tmp/m4_daily_0.1.0.zip'
```
Comment (besirkurtulmus) [inline src/OpenAnomalyDetection_test.py]: Is this commented line needed?

Comment (zeryx) [reply src/OpenAnomalyDetection_test.py]: you're right, I'm actually going to create an "algorithmia compatible" and another version that's entirely local - that will be a cleaner experience.

**Review by kennydaniel — APPROVED** [review]
Overall looks like an improvement, a lot of changes though. In the future maybe try to split into smaller chunks (its already split across several commits that could maybe have been distinct PRs)

"Show a little code, get a lot of feedback. Show a lot of code, get only a little feedback"

**Comment by zeryx** [comment]
That's good advice, I'll look to figure out how to break up in the future. My git-fu needs work

**Review by j-vasanth — APPROVED** [review]
LGTM overall. It'd be nice to see some documentation in the main python src. For all functions and for describing the flow in the apply function.

I agree with Kenny on breaking PR into smaller chunks, perhaps in different branches.

**Comment by zeryx** [comment]
Docs are in the next PR. All major functions will be described and two readmes will be built out. Yea I need to figure out a good pipeline for this kind of work - how can I split 1 pr into 10 cleanly?

**Comment by kennydaniel** [comment]
Since this one has already been reviewed, I would just leave it as is. In the future just be more mindful of smaller PRs.

If you already have a PR you're trying to split, its possible to do some git magic like rebaseing and cherry-picking, but usually easier to just keep each branch to the narrowest feature you can get to compile on its own.

**Pull request merged**